### PR TITLE
Cleanup calendar code duplication

### PR DESCRIPTION
**Pull request by pentp**

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs [hunk 1]
```diff
@@ -480,15 +373,13 @@ public int[] Eras
         {
             get
             {
-                if (m_eras == null)
+                EraInfo[] eraInfo = m_EraInfo;
```
Comment (tarekgh) [inline src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs:376]: >                 [](http://example.com/codeflow?start=0&length=16)

why got rid of m_eras and filling the returning array every time the property get called?

**Comment by tarekgh** [comment]
@pentp thanks for this clean up. LGTM. I added a little comment.

File: src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs (hunk 1 above)
Comment (pentp) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs:376]: Is array cloning significantly cheaper for such small arrays?

Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs:376]: Array.Clone usually will use Buffer.Memmove for the arrays not having GC pointers (like the case here). I believe that would be faster, but I didn't measure it to give you how much would be faster.

Comment (pentp) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs:376]: A quick benchmark shows that for a 5 element array the cached version is about 10 times slower:
|        Method |     Mean |    Error |   StdDev |
|-------------- |---------:|---------:|---------:|
| GetErasCached | 97.86 ns | 0.999 ns | 0.594 ns |
|       GetEras | 10.93 ns | 1.007 ns | 0.666 ns |

Comment (tarekgh) [reply src/libraries/System.Private.CoreLib/src/System/Globalization/GregorianCalendarHelper.cs:376]: Interesting. That is very surprising. Anyway, thanks for measuring it. we should be good then.

**Review by tarekgh — APPROVED** [review]

**Pull request merged**